pkg/virt-launcher/virtwrap/network/podinterface.go

```diff
@@ -706,10 +710,33 @@ func configureVifV6Addresses(p *MasqueradePodInterface, err error) error {
 	return nil
 }
 
-func (p *MasqueradePodInterface) startDHCP(vmi *v1.VirtualMachineInstance) error {
+func (p *MasqueradePodInterface) startDynamicIPServers(vmi *v1.VirtualMachineInstance) error {
+	if err := p.startRADaemon(); err != nil {
+		log.Log.Criticalf("could not start the RA daemon: %v", err)
+		return err
+	}
 	return Handler.StartDHCP(p.vif, p.vif.Gateway, p.bridgeInterfaceName, p.iface.DHCPOptions)
 }
 
+func (p *MasqueradePodInterface) startRADaemon() error {
+	theBridge, err := Handler.LinkByName(p.bridgeInterfaceName)
+	if err != nil {
+		return err
+	}
+
+	targetPID := "self"
+	if err := Handler.CreateRADaemon(
+		GetNDPConnectionUnixSocketPath(targetPID, p.bridgeInterfaceName),
+		p.bridgeInterfaceName,
+		api.DefaultVMIpv6CIDR,
```

Comment:
I should probably get this half-ready for having a configurable CIDR.

I think I need to cache this - however, we're just caching the pod interface, and the guest (VM) interface. No precedent for the bridge interface. Should I create a new cache file, of just plug this into one of the existing interface caches ? 

@phoracek / @EdDev / @AlonaKaplan

Reply:
This is already stored in - vif.IPv6.Mask

Reply:
I don't follow; shouldn't I simply use `b.vmIpv6NetworkCIDR` ? It is set in https://github.com/kubevirt/kubevirt/blob/master/pkg/virt-launcher/virtwrap/network/podinterface.go#L702 , as part of the `discoverPodNetworkInterface` operation.

Reply:
`startRADaemon` is invoked in phase2, `b.vmIpv6NetworkCIDR` is not initialized in this phase.
It is indeed confusing. We should remove `vmIpv6NetworkCIDR` from the `MasqueradeBindMechanism`.

Reply:
Done in https://github.com/kubevirt/kubevirt/pull/4581/commits/a8c01317d2ac6d349d250a1a8ae549749f0c7b75